tensor/shape.go

```diff
@@ -22,7 +22,9 @@ func (s Shape) TotalSize() int {
 	return ProdInts([]int(s))
 }
 
-func (s Shape) calcStrides() []int {
+// calcStrides has an additional optional argument, maks. It is used to mask out given dimensions
```

Comment:
Sounds fine. I tried to stay away from long names but I agree with you that it is best to minimize optional arguments and empty interfaces, makes things less likely to break.